docs/serviceprincipal.md

```diff
@@ -25,12 +25,32 @@ There are several ways to create a Service Principal in Azure Active Directory:
    az ad sp create-for-rbac --role="Contributor" --scopes="/subscriptions/${SUBSCRIPTION_ID}"
    ```
 
-This will output your `client_id`, `client_secret` (password), `sp_name`, and `tenant`.  The `sp_name` or `client_id` may be used for the `servicePrincipalProfile.servicePrincipalClientId` and the `client_secret` is used for `servicePrincipalProfile.servicePrincipalClientSecret`.
+You can set `SUBSCRIPTION_ID` like this:
+
+```
+SUBSCRIPTION_ID=$(az account list --query "[?name == '<i>subscription name</i>'].[id]" --out tsv)
+```
+
+With CLI output format set to JSON, the output will look something like this:
+
+```
+{
+  "appId": "<i>some GUID</i>",
+  "name": "http://azure-cli-2016-<i>more characters</i>",
+  "password": "<i>password GUID</i>",
+  "tenant": "<i>tenant GUID</i>"
+}
+
+```
+
+The `appId` is the Azure AD `client_id`, the `password' is the `client_secret`, `the `name` is the `service principal name`, and `tenant` is the GUID identifiying the tenant in which the service principal was created.  The `service principal name` or `client_id` may be used for the `servicePrincipalProfile.servicePrincipalClientId` and the `password` or `client_secret` is used for `servicePrincipalProfile.servicePrincipalClientSecret`.

```

Comment:
If you wanted to file an issue on azure/azure-cli, that'd be great. @yugangw-msft would be the person to tag.